### PR TITLE
staq 3.0.1 (new formula)

### PR DESCRIPTION
**Pull request by vsoftco**
Full-stack quantum processing toolkit

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/staq.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  desc "Full-stack quantum processing toolkit"
+  homepage "https://github.com/softwareQinc/staq"
+  url "https://github.com/softwareQinc/staq/archive/v3.0.1.tar.gz"
+  sha256 "306e46e3a8517b92c58b06be1ce78d104c112c2f0f7a7dde37a75d1f934f0c4c"
+  license "MIT"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
```
Comment (carlocab) [inline Formula/staq.rb:11]: Do you need this? i.e. does `brew livecheck staq` return wrong results without this?

**Review by carlocab — COMMENTED** [review]
Thanks; a few comments. Please also squash your commits together into your first commit (including after making any changes here).

File: Formula/staq.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      OPENQASM 2.0;
+      include "qelib1.inc";
+      qreg q[1];
+      creg c[1];
+      h q[0];
+      h q[0];
+      measure q->c;
+    EOS
+    assert_equal "OPENQASM 2.0; include \"qelib1.inc\"; qreg q[1]; creg c[1]; measure q[0] -> c[0];",
+    shell_output("#{bin}/staq -O3 ./input.qasm").squish
```
Comment (carlocab) [inline Formula/staq.rb:32]: Does this work?
```suggestion
    assert_equal <<~EOS, shell_output("#{bin}/staq -O3 ./input.qasm")
      OPENQASM 2.0;
      include "qelib1.inc";
      qreg q[1];
      creg c[1];
      measure q[0] -> c[0];
    EOS
```

File: Formula/staq.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      qreg q[1];
+      creg c[1];
+      h q[0];
+      h q[0];
+      measure q->c;
+    EOS
+    assert_equal <<~EOS, shell_output("#{bin}/staq -O3 ./input.qasm")
+      OPENQASM 2.0;
+      include "qelib1.inc";
+      qreg q[1];
```
Comment (carlocab) [inline Formula/staq.rb:34]:
```suggestion
    assert_equal <<~EOS.strip, shell_output("#{bin}/staq -O3 ./input.qasm").strip
      OPENQASM 2.0;
      include "qelib1.inc";

      qreg q[1];
```

**Comment by vsoftco** [comment]
@carlocab Should be ready to go (after tests pass), thanks!

**Review by carlocab — DISMISSED** [review]

**Comment by vsoftco** [comment]
Still got an error in the test, will fix asap

**Review by carlocab — DISMISSED** [review]

**Comment by vsoftco** [comment]
@carlocab Thank you very much for all your help. Looks like all passes, didn't know I should give access to homebrew on my local branch, so looks like you need to re-approve the replaced commit :)

**Pull request closed without merging**